com.ibm.streamsx.hdfs/impl/java/src/com/ibm/streamsx/hdfs/HDFS2FileSink.java

```diff
@@ -755,7 +789,7 @@ public void processPunctuation(StreamingInput<Tuple> arg0, Punctuation punct)
 		}
 		// set the file to expire after punctuation
 		// on the next write, the file will be recreated
-		else if (isCloseOnPunct() && fFileToWrite != null) {
```

Comment:
Why did you get rid of the fFileToWrite != null? 

This query is now done in method closeFile() thus there is no need to make 
this query here.
But it seems that the trace log in line 820 is redundant. It should be 
moved after the query if ( ! alreadyClosed) it should have the old tesxt '
Set file as expired '